EngineLayer/FdrAnalysis/PEPValueAnalysisGeneric.cs

```diff
@@ -170,14 +170,21 @@ private static int GetChargeStateMode(List<PeptideSpectralMatch> psms)
                 Dictionary<int, List<double>> hydrobophobicites = new Dictionary<int, List<double>>();
                 Dictionary<int, Tuple<double, double>> averagesCommaStandardDeviations = new Dictionary<int, Tuple<double, double>>();
 
-                foreach (PeptideSpectralMatch psm in psms.Where(f => (f.FullFilePath == filename || f.FullFilePath == null) && f.FdrInfo.QValue <= 0.01))
+                foreach (PeptideSpectralMatch psm in psms.Where(f => (f.FullFilePath == filename || f.FullFilePath == null) && f.FdrInfo.QValue <= 0.01 && !f.IsDecoy))
                 {
+                    List<string> fullSequences = new List<string>();
                     foreach ((int notch, PeptideWithSetModifications pwsm) in psm.BestMatchingPeptides)
                     {
-                        if (pwsm.AllModsOneIsNterminus.Any() && !computeHydrophobicitiesforModifiedPeptides)
+                        if (fullSequences.Contains(pwsm.FullSequence))
+                        {
+                            continue;
+                        }
+                        fullSequences.Add(pwsm.FullSequence);
+                        double predictedHydrophobicity = calc.ScoreSequence(pwsm);
+                        int possibleKey = (int)(2 * Math.Round(psm.ScanRetentionTime / 2d, 0));
```

Comment:
int possibleKey = (int)(2 * Math.Round(psm.ScanRetentionTime / 2d, 0)):
Could you add annotation/explain?